File: Programmars/Level_1/정수 내림차순으로 배치하기.py
def solution(n):
    answer = ''
    for x in sorted(list(str(n)), reverse=True):
        answer += x
    return int(answer)


print(solution(118372))  # 873211


# ref: join() 활용
def solution(n):
    ls = list(str(n))
    ls.sort(reverse=True)
    return int("".join(ls))

File: JavaScript/Programmars/Level_1/정수 내림차순으로 배치하기.js
// 내 풀이
function solution(n) {
  const numArr = String(n)
    .split('')
    .map((val, idx) => {
      return Number(val);
    });
  numArr.sort((x, y) => y - x);

  return Number(numArr.join(''));
}

console.log(solution(118372)); // 873211

// 다른 풀이
function solution(n) {
  const newN = n + '';
  const newArr = newN.split('').sort().reverse().join('');

  return +newArr;
}
